aten/src/ATen/native/cuda/FractionalMaxPool3d.cu

```diff
@@ -167,10 +167,16 @@ void fractional_max_pool3d_out_cuda_template(
 
     int64_t ndims = input.ndimension();
     TORCH_CHECK(
-      input.numel() != 0 && (ndims == 4 || ndims == 5),
-      "fractional_max_pool3d_out_cuda_template(): ",
-      "non-empty 4D or 5D (batch mode) tensor expected for input, but got: ",
-      ndims);
+                ndims == 4 || ndims == 5,
+                "fractional_max_pool3d_out_cuda_template(): ",
+                "Expected 4D or 5D tensor with optional 0-dim batch size, but got: ",
```

Comment:
nit: I think since this check is only concerned with the number of dims, "fractional_max_pool2d(): Expected 3D or 4D tensor, but got: " is better